app/components/ui/breadcrumb/component.rb

```diff
@@ -0,0 +1,26 @@
+module UI
+  module Breadcrumb
+    class Component < ApplicationComponent
+      style do
+        base { %w[my-6 font-medium flex text-zinc-500 text-sm] }
+      end
+
+      style :ol do
+        base { %w[flex flex-wrap items-center space-x-3 list-none m-0 p-0] }
+      end
+
+      renders_many :items, UI::Breadcrumb::ItemComponent
```

Comment:
Uma outra estratégia seria considerar links e separators como tipos de items. Então aqui usaria o recurso de [polymorphic slots](https://viewcomponent.org/guide/slots.html#polymorphic-slots) do view components. Ficaria algo como:

```suggestion
      renders_many :items, types: {
            link: { renders: UI::Breadcrumb::LinkComponent, as: :link },
            separator { renders: UI::Breadcrumb::SeparatorComponent, as: :separator },
      }
```

As classes que pertencem ao `ItemComponent` seriam movidas para link e separators para a estrutura e estilização deles fosse igual ao que já existe.

A principal diferença dessa estratégia é que a renderização do componente fica assim:
```erb
<%= render UI::Breadcrumb::Component.new do |breadcrumb| %>
    <% breadcrumb.with_link(href: "#", first: true).with_content("Home") %>
    <% breadcrumb.with_separator %>
    <% breadcrumb.with_link(href: "#").with_content("Library") %>
    <% breadcrumb.with_separator %>
    <% breadcrumb.with_link(href: "#", active: true).with_content("Data") %>
<% end %>
```

Reply:
boa! vai facilitar bastante \o/